### PR TITLE
Topic/anyhow

### PR DESCRIPTION
**Pull request by d-e-s-o**

### CODE REVIEW

**Comment by d-e-s-o** [comment]
Here is the patch set for moving to `anyhow` for error handling. @robinkrahl, may I ask you to take a look? I guess mostly high level. I still haven't given it the final polish (or looked at the test/lint failures in detail), but will do so before check in.

I'd actually want to merge this work next and only then come to the config work you have done (which I understand also is not quite ready because of a crate not yet being published), mostly because I'd want to avoid rebasing all this `.context(...)` stuff and reevaluating call sites again, as I had already gone over all of that. I can take care of rebasing your pull request and making it work a bit more nicely with the changed error handling model. Would that be okay with you?

File: Cargo.lock [hunk 1]
```diff
@@ -9,6 +9,12 @@ dependencies = [
  "memchr",
 ]
 
+[[package]]
+name = "anyhow"
+version = "1.0.28"
```
Comment (robinkrahl) [inline Cargo.lock:14]: Could be updated to 1.0.32.

File: src/args.rs [hunk 2]
```diff
@@ -134,13 +134,13 @@ pub enum ConfigOption<T> {
 }
 
 impl<T> ConfigOption<T> {
-  pub fn try_from(disable: bool, value: Option<T>, name: &'static str) -> Result<Self, String> {
+  pub fn try_from(disable: bool, value: Option<T>, name: &'static str) -> anyhow::Result<Self> {
     if disable {
       if value.is_some() {
-        Err(format!(
+        anyhow::bail!(
```
Comment (robinkrahl) [inline src/args.rs:140]: `anyhow::ensure!(value.is_none(), ...)` might be easier to read.

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,6 +1,8 @@
 Unreleased
 ----------
 - Changed default OTP format from `hex` to `base32`
+- Updated minimum required Rust version to `1.42.0`
+- Bumped `nitrokey` dependency to `0.7.1`
```
Comment (robinkrahl) [inline CHANGELOG.md:5]: List the anyhow dependency and the error refactoring?

**Review by robinkrahl — APPROVED** [review]
LGTM.  Thank you!

**Comment by d-e-s-o** [comment]
Thanks for your review, Robin! Everything is in `devel` now.

**Pull request closed without merging**